tests/utils.js

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs');
+const path = require('path');
+const {google} = require('googleapis');
+
+const pathToServiceAccount = path.resolve(__dirname, '../resources/serviceAccount.json');
+if (!fs.existsSync(pathToServiceAccount)) {
+  console.log(
+    `[ERROR] Firebase service account not found. Please place it in ${pathToServiceAccount}.`
+  );
+  process.exit(1);
+}
+
+const serviceAccount = require(pathToServiceAccount);
+
+/**
+ * Generates a Google OAuth2 access token for use authenticating to Firebase services, such as the
+ * Firebase Realtime Database REST API.
+ * 
+ * See https://firebase.google.com/docs/database/rest/auth#google_oauth2_access_tokens.
+ */
+module.exports.fetchAccessToken = () => {
```

Comment:
I considered packaging this method with the library itself, but we need to be able to access `client_email` and `private_key`. We can actually get to both by importing `firebase-admin` and then doing `admin.app().options.credential.certificate_.clientEmail` and `admin.app().options.credential.certificate_.privateKey`, but that is super hacky and almost certain to break at some point. Unfortunately, Firebase doesn't really have an API to unpack the credential the API is authenticated with and (more importantly) no API to fetch the Google OAuth2 access tokens it itself uses.

So, ultimately, I think the calling code should continue to provide the credential, even though it's a bit more complicated to make than it was previously.